### PR TITLE
Intern `util.Address` strings

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: pkg/process/checks/net.go [hunk 1]
```diff
@@ -106,8 +106,9 @@ func (c *ConnectionsCheck) getConnections() ([]*model.Connection, error) {
 		}
 		cs, err := c.localTracer.GetActiveConnections(c.tracerClientID)
 		conns := make([]*model.Connection, len(cs.Conns))
+		cache := make(ebpf.AddrCache)
```
Comment (p-lambert) [inline pkg/process/checks/net.go]: This is kinda of ugly (should be hidden from clients) but once we get rid of the local system-probe bth `FormatConnection` and `AddrCache` will become private and be called internally from the marshalers

File: pkg/ebpf/encoding/format.go [hunk 2]
```diff
@@ -7,12 +7,26 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/process/util"
 )
 
+// AddrCache is responsible for interning util.Address strings
+type AddrCache map[util.Address]string
```
Comment (p-lambert) [inline pkg/ebpf/encoding/format.go]: See comment below

Comment (sunhay) [inline pkg/ebpf/encoding/format.go]: Do we ever reset this?

Comment (p-lambert) [reply pkg/ebpf/encoding/format.go]: As it is it doesn't grow unbounded (it's not used globally but within a single `Marshal` function call)

**Pull request closed without merging**